Test Failure (20200925-1708): junit.framework.TestSuite.com.ibm.ws.logstash.collector.tests.LogstashSSLTest

RTC 278208

```
org.testcontainers.containers.ContainerLaunchException: Container startup failed
at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:290)
at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:272)
at org.testcontainers.containers.GenericContainer.starting(GenericContainer.java:807)
at org.testcontainers.containers.FailureDetectingExternalResource$1.evaluate(FailureDetectingExternalResource.java:29)
at componenttest.custom.junit.runner.FATRunner$2.evaluate(FATRunner.java:318)
at componenttest.custom.junit.runner.FATRunner.run(FATRunner.java:171)
Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception
at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:83)
at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:283)
Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container
at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:350)
at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:285)
at org.testcontainers.containers.GenericContainer$$Lambda$127/0000000000000000.call(Unknown Source)
at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:76)
Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for container port to open (libr-docker-05.fyre.ibm.com ports: [49642] should be listening)
at org.testcontainers.containers.wait.strategy.HostPortWaitStrategy.waitUntilReady(HostPortWaitStrategy.java:47)
at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:35)
at org.testcontainers.containers.wait.HostPortWaitStrategy.waitUntilReady(HostPortWaitStrategy.java:23)
at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:35)
at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:675)
at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:332)
```

--- a/dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/LogstashCollectorTest.java
+++ b/dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/LogstashCollectorTest.java
@@ -15,6 +15,7 @@ import static org.junit.Assert.assertNotNull;
 import java.io.File;
 import java.io.UnsupportedEncodingException;
 import java.net.URLEncoder;
+import java.time.Duration;
 import java.util.ArrayList;
 import java.util.Iterator;
 import java.util.List;
@@ -185,6 +186,7 @@ public abstract class LogstashCollectorTest {
                     .withFileFromFile("/usr/share/logstash/config/logstash.key", new File(PATH_TO_AUTOFVT_TESTFILES + "logstash.key"), 644) //
                     .withFileFromFile("/usr/share/logstash/config/logstash.crt", new File(PATH_TO_AUTOFVT_TESTFILES + "logstash.crt"), 644)) //
                                     .withExposedPorts(5043) //
+                                    .withStartupTimeout(Duration.ofSeconds(90)) //
                                     .withLogConsumer(LogstashCollectorTest::log); //
 
     // This helper method is passed into `withLogConsumer()` of the container

--- a/dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/LogstashSSLTest.java
+++ b/dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/LogstashSSLTest.java
@@ -158,7 +158,7 @@ public class LogstashSSLTest extends LogstashCollectorTest {
 
         createMessageEvent(testName);
 
-        assertNotNull("Cannot find TRAS0218I from messages.log", server.waitForStringInLogUsingMark("TRAS0218I", 10000));
+        assertNotNull("Cannot find TRAS0218I from messages.log", server.waitForStringInLogUsingMark("TRAS0218I"));
         assertNotNull("Did not find " + LIBERTY_MESSAGE, waitForStringInContainerOutput(LIBERTY_MESSAGE));
     }
 
@@ -170,7 +170,7 @@ public class LogstashSSLTest extends LogstashCollectorTest {
 
         createAccessLogEvent(testName);
 
-        assertNotNull("Cannot find TRAS0218I from messages.log", server.waitForStringInLogUsingMark("TRAS0218I", 10000));
+        assertNotNull("Cannot find TRAS0218I from messages.log", server.waitForStringInLogUsingMark("TRAS0218I"));
         assertNotNull("Did not find " + LIBERTY_ACCESSLOG, waitForStringInContainerOutput(testName));
     }
 
@@ -207,7 +207,7 @@ public class LogstashSSLTest extends LogstashCollectorTest {
         createFFDCEvent(3);
         Log.info(c, testName, "------> finished ffdc3(ArrayIndexOutOfBoundsException)");
         exceptions.add("ArrayIndexOutOfBoundsException");
-        assertNotNull("Cannot find TRAS0218I from messages.log", server.waitForStringInLogUsingMark("TRAS0218I", 10000));
+        assertNotNull("Cannot find TRAS0218I from messages.log", server.waitForStringInLogUsingMark("TRAS0218I"));
 
         assertNotNull(LIBERTY_FFDC + " not found", waitForStringInContainerOutput(LIBERTY_FFDC));
         assertNotNull("ArithmeticException not found", waitForStringInContainerOutput("ArithmeticException"));

--- a/dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/LogstashSSLTest.java
+++ b/dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/LogstashSSLTest.java
@@ -429,8 +429,8 @@ public class LogstashSSLTest extends LogstashCollectorTest {
             }
             Log.info(c, testName, " jar file for health center under path " + JAVA_HOME + "/lib/ext/healthcenter.jar exist:"
                                   + new File(JAVA_HOME + "/lib/ext/healthcenter.jar").exists());
-        } else if (JAVA_HOME.endsWith("bin")) {
-            healthCenterInstalled = findHealthCenterDirecotry(JAVA_HOME.substring(0, JAVA_HOME.indexOf("bin") + 1));
+        } else if ((JAVA_HOME.endsWith("/bin")) || (JAVA_HOME.endsWith("\\bin"))) {
+            healthCenterInstalled = findHealthCenterDirecotry(JAVA_HOME.substring(0, JAVA_HOME.length() - 4));
             if (!healthCenterInstalled) {
                 Log.info(c, testName, " unable to find heathcenter.jar, thus unable to produce gc events. Thus, this check will be by-passed");
             }
@@ -449,19 +449,24 @@ public class LogstashSSLTest extends LogstashCollectorTest {
 
     private boolean findHealthCenterDirecotry(String directoryPath) {
         boolean jarFileExist = false;
-        File[] files = new File(directoryPath).listFiles();
-        for (File file : files) {
-            if (file.isDirectory()) {
-                jarFileExist = findHealthCenterDirecotry(file.getAbsolutePath());
-                if (jarFileExist == true) {
-                    return true;
-                }
-            } else {
-                if (file.getAbsolutePath().contains("healthcenter.jar")) {
-                    Log.info(c, testName, " healthcetner.jar is found under path " + file.getAbsolutePath());
-                    return true;
+        File dirFile = new File(directoryPath);
+        if (dirFile.exists()) {
+            File[] files = dirFile.listFiles();
+            for (File file : files) {
+                if (file.isDirectory()) {
+                    jarFileExist = findHealthCenterDirecotry(file.getAbsolutePath());
+                    if (jarFileExist == true) {
+                        return true;
+                    }
+                } else {
+                    if (file.getAbsolutePath().contains("healthcenter.jar")) {
+                        Log.info(c, testName, " healthcetner.jar is found under path " + file.getAbsolutePath());
+                        return true;
+                    }
                 }
             }
+        } else {
+            Log.info(c, "findHealthCenterDirecotry", "directoryPath " + directoryPath + " does not exist");
         }
         return jarFileExist;
     }

--- a/dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/ThrottleMaxEventsTest.java
+++ b/dev/com.ibm.ws.logstash.collector_fat/fat/src/com/ibm/ws/logstash/collector/tests/ThrottleMaxEventsTest.java
@@ -107,7 +107,6 @@ public class ThrottleMaxEventsTest extends LogstashCollectorTest {
             createTraceEvent(Integer.toString(t));
         }
         waitForContainerOutputSize(400);
-        //wait for the message events to process total of 400 events
         boolean throttled = waitForEventsToProcess(100, 100, 200, startTime, 120000);
 
         Log.info(c, testName, "Throttling test=" + throttled);
@@ -159,24 +158,38 @@ public class ThrottleMaxEventsTest extends LogstashCollectorTest {
         long traceEndTime = 0;
         long messageEndTime = 0;
         long accesslogEndTime = 0;
-        List<JSONObject> jobjs = parseJsonInContainerOutput();
-        for (JSONObject jobj : jobjs) {
-            String type = jobj.getString(KEY_TYPE);
-            String message = null;
-            if (!type.equals(LIBERTY_ACCESSLOG)) {
-                message = jobj.getString(KEY_MESSAGE);
+        while ((accessCount + traceCount + messageCount < numMsg + numTrace + numAccessLog) && (timeout > 0)) {
+            try {
+                Thread.sleep(1000);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
             }
-            if (type.equals(LIBERTY_TRACE) && message.contains("TEST JUL TRACE")) {
-                traceCount++;
-                traceEndTime = System.currentTimeMillis();
-            } else if (type.equals(LIBERTY_MESSAGE) && message.contains("Test Logstash Message")) {
-                messageCount++;
-                messageEndTime = System.currentTimeMillis();
-            } else if (type.equals(LIBERTY_ACCESSLOG)) {
-                accessCount++;
-                accesslogEndTime = System.currentTimeMillis();
+            timeout -= 1000;
+
+            List<JSONObject> jobjs = parseJsonInContainerOutput();
+            messageCount = 0;
+            traceCount = 0;
+            accessCount = 0;
+
+            for (JSONObject jobj : jobjs) {
+                String type = jobj.getString(KEY_TYPE);
+                String message = null;
+                if (!type.equals(LIBERTY_ACCESSLOG)) {
+                    message = jobj.getString(KEY_MESSAGE);
+                }
+                if (type.equals(LIBERTY_TRACE) && message.contains("TEST JUL TRACE")) {
+                    traceCount++;
+                    traceEndTime = System.currentTimeMillis();
+                } else if (type.equals(LIBERTY_MESSAGE) && message.contains("Test Logstash Message")) {
+                    messageCount++;
+                    messageEndTime = System.currentTimeMillis();
+                } else if (type.equals(LIBERTY_ACCESSLOG)) {
+                    accessCount++;
+                    accesslogEndTime = System.currentTimeMillis();
+                }
             }
         }
+
         assertEquals("Number of liberty_message does not match", numMsg, messageCount);
         assertEquals("Number of liberty_trace does not match", numTrace, traceCount);
         assertEquals("Number of liberty_accesslog does not match", numAccessLog, accessCount);

--- a/dev/com.ibm.ws.logstash.collector_fat/publish/files/logstash.yml
+++ b/dev/com.ibm.ws.logstash.collector_fat/publish/files/logstash.yml
@@ -1,1 +1,2 @@
 xpack.monitoring.enabled: false
+pipeline.batch.size: 1

